File: graal-java-graphql/src/main/java/lan/test/graal/TypeDataFetcher.java
package lan.test.graal;

import graphql.schema.DataFetcher;
import graphql.schema.DataFetchingEnvironment;

import javax.inject.Singleton;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

@Singleton
public class TypeDataFetcher implements DataFetcher<List<Type>> {
    public List<Type> get(DataFetchingEnvironment environment) {
        return Collections.singletonList(Type.DEFAULT_TYPE);
    }

}

File: graal-kotlin-graphql/src/main/kotlin/lan/test/graal/TypeDataFetcher.kt
package lan.test.graal

import graphql.schema.DataFetcher
import graphql.schema.DataFetchingEnvironment
import lan.test.graal.Type
import javax.inject.Singleton

@Singleton
class TypeDataFetcher: DataFetcher<List<Type>> {
    override fun get(environment: DataFetchingEnvironment?): List<Type> {
        return listOf(Type.DEFAULT_TYPE)
    }

}
